Update POS to POS server for requests to centrapay

alexBinns opened the pull request:
There was some feedback from an integrator that they were confused with the naming and thought that their POS should be making the API calls to Centrapay. 
This PR updates the references from POS to POS server where the entity is making API calls to Centrapay.

![image](https://github.com/user-attachments/assets/c1832b12-b1e7-4ea4-b857-954adec188df)
![image](https://github.com/user-attachments/assets/3b139c56-1c55-4f8b-87ce-468312497aa8)
![image](https://github.com/user-attachments/assets/45b3c573-ef0d-4856-a735-95c2ecf3547b)
![image](https://github.com/user-attachments/assets/312b8ae7-bb22-45b7-a1da-64d22acfdf5f)
![image](https://github.com/user-attachments/assets/58633c85-6d13-457a-a45d-c269876b1357)
![image](https://github.com/user-attachments/assets/1c872f20-8c25-4719-bc87-0be05c533c4c)
![image](https://github.com/user-attachments/assets/81b50079-eefc-4ba5-a497-2ffa800fb73d)


Testing Steps:
* Check updated pages have been updated.

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -34,21 +34,20 @@ sequenceDiagram
 
 	participant Cardholder
 	participant POS
```
should we remove POS?

alexBinns replied:
No we want POS as well so we can distinguish between what should happen on the front end and what should happen on the backend

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -34,21 +34,20 @@ sequenceDiagram
 	participant Centrapay
 
 	Note over Cardholder: Present Barcode
 	Cardholder->>POS: Scan Barcode
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -34,21 +34,20 @@ sequenceDiagram
 
-		alt Payment Condition Pending
+		opt Payment Condition Pending
 			note over POS: ❓ Prompt Cashier to Accept or Decline Condition
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -382,41 +382,40 @@ autonumber
 
 participant Cardholder
 participant POS
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -382,41 +382,40 @@ autonumber
 participant Centrapay
 
 Note over Cardholder: Present Barcode
 Cardholder->>POS: Scan Barcode
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -382,41 +382,40 @@ autonumber
-			Note over POS: Stop Polling for Confirmation
-		end
+  opt Payment Condition Pending
+    note over POS: ❓ Prompt Cashier to Accept or Decline Condition
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -382,41 +382,40 @@ autonumber
-		end
+  opt Payment Condition Pending
+    note over POS: ❓ Prompt Cashier to Accept or Decline Condition
+    POS->>Centrapay: POS Accepts Condition
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -382,41 +382,40 @@ autonumber
-	POS->>Centrapay: Make Confirmations Against Authorised Funds
-
+	POS Server->>Centrapay: Make Confirmations Against Authorised Funds
 	Note over POS: ✅ Display Confirmation
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
````diff
@@ -382,41 +382,40 @@ autonumber
+end
 ```
 
 1. The Cardholder presents their Farmlands Card for the POS to scan.
````
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -573,7 +572,8 @@ Note over POS: ✅ Display Release confirmation
 
-4. [Polling the Payment Request](/api/payment-requests#get-a-payment-request) returns a list of pending `merchantConditions` in the response. The POS must prompt the cashier to [accept](/api/payment-requests#accept-a-payment-condition) or [decline](/api/payment-requests#decline-a-payment-condition) each [Payment Condition](#payment-conditions) in order to successfully authorise a Pre Auth payment ...
+4. [Polling the Payment Request](/api/payment-requests#get-a-payment-request) returns a list of pending `merchantConditions` in the response.
+   The POS must prompt the cashier to [accept](/api/payment-requests#accept-a-payment-condition) or [decline](/api/payment-requests#decline-a-payment-condition) each [Payment Condition](#payment-conditions) in order to successfully authorise a Pre Auth payment.
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -846,19 +848,20 @@ sequenceDiagram
 
 	participant Cardholder
 	participant POS
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -846,19 +848,20 @@ sequenceDiagram
+	participant POS Server
 	participant Centrapay
 
 	Cardholder->>POS: Present Transaction Reference
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -846,19 +848,20 @@ sequenceDiagram
 
 	Cardholder->>POS: Present Transaction Reference
 
 	Note over POS: Look up Payment Request ID
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -846,19 +848,20 @@ sequenceDiagram
-	POS->>Centrapay: Refund Payment Request
+	POS Server->>Centrapay: Refund Payment Request
 
 	Note over POS: ✅ Display Refund Confirmation
```
POS

pupato13 commented on src/content/guides/farmlands-pos-integration.mdoc:
````diff
@@ -846,19 +848,20 @@ sequenceDiagram
 	Note over POS: ✅ Display Refund Confirmation
 ```
 
 1. The Cardholder presents a transaction reference on their paper receipt. The POS looks up the [Payment Request](/api/payment-requests/) ID using the transaction reference.
````
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
```diff
@@ -18,16 +18,17 @@ sequenceDiagram
 
 	participant Patron
 	participant POS
```
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
```diff
@@ -18,16 +18,17 @@ sequenceDiagram
 	participant Centrapay
 
 	Note over Patron: Present Barcode
 	Patron->>POS: Scan Barcode
```
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
````diff
@@ -38,7 +39,7 @@ sequenceDiagram
 ```
 
 1. The patron presents a Barcode for the POS to scan.
````
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
```diff
@@ -200,24 +202,25 @@ sequenceDiagram
 
 	participant Patron
 	participant POS
```
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
```diff
@@ -200,24 +202,25 @@ sequenceDiagram
 	participant Centrapay
 
 	Note over Patron: Present Barcode
 	Patron->>POS: Scan Barcode
```
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
```diff
@@ -200,24 +202,25 @@ sequenceDiagram
+		note over Centrapay: Confirm Details and Complete Payment
 	end
 
 	Note over POS: ✅ Display Successful Payment
```
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
````diff
@@ -200,24 +202,25 @@ sequenceDiagram
 	Note over Patron: ✅ Display Successful Payment
 ```
 
 1. The patron presents a Barcode for the POS to scan.
````
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
```diff
@@ -335,27 +339,29 @@ sequenceDiagram
 	autonumber
 
 	participant Patron
 	participant POS
```
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
```diff
@@ -335,27 +339,29 @@ sequenceDiagram
 	participant Centrapay
 
 	Note over Patron: Present Barcode
 	Patron->>POS: Scan Barcode
```
POS

pupato13 commented on src/content/guides/merchant-integration-barcode-flow.mdoc:
````diff
@@ -335,27 +339,29 @@ sequenceDiagram
+	POS Server->>Centrapay: Create Payment Request with Barcode
 ```
 
 1. The patron presents a barcode for the POS to scan.
````
POS

pupato13 commented on src/content/guides/merchant-integration-qr-code-flow.mdoc:
```diff
@@ -16,17 +16,18 @@ sequenceDiagram
 
 	participant Patron
 	participant POS
```
POS

pupato13 commented on src/content/guides/merchant-integration-qr-code-flow.mdoc:
```diff
@@ -16,17 +16,18 @@ sequenceDiagram
-	POS->>Centrapay: Create Payment Request
+	POS Server->>Centrapay: Create Payment Request
 
 	note over POS: Present QR Code
```
POS

pupato13 commented on src/content/guides/merchant-integration-qr-code-flow.mdoc:
```diff
@@ -16,17 +16,18 @@ sequenceDiagram
 
 	note over POS: Present QR Code
 
 	Patron->>POS: Scan QR Code
```
POS

pupato13 commented on src/content/guides/merchant-integration-qr-code-flow.mdoc:
```diff
@@ -36,7 +37,8 @@ sequenceDiagram
 
-1. The POS [creates a Payment Request](/api/payment-requests#create-a-payment-request) and presents a QR code to the Patron on a customer-facing display.
+1. The POS server [creates a Payment Request](/api/payment-requests#create-a-payment-request).
+   The POS then presents a QR code to the Patron on a customer-facing display.
```
POS

MeganSteenkamp commented:
@alexBinns would be good if you could put screenshots of the diagrams next time please 🙌

MeganSteenkamp commented:
@alexBinns Change: I don't think the diagrams are correct. We usually have a continuous flow of arrows whereas you have a gap between POS and POS server. You are missing communication between the two.
![Screenshot 2025-03-25 at 8 32 55 am](https://github.com/user-attachments/assets/ba03a73b-829d-4b1a-9fd0-9beec328706a)

alexBinns commented:
> @alexBinns Change: I don't think the diagrams are correct. We usually have a continuous flow of arrows whereas you have a gap between POS and POS server. You are missing communication between the two. ![Screenshot 2025-03-25 at 8 32 55 am](https://private-user-images.githubusercontent.com/64108933/426272303-ba03a73b-829d-4b1a-9fd0-9beec328706a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwMTkyMjUsIm5iZiI6MTc0NDAxODkyNSwicGF0aCI6Ii82NDEwODkzMy80MjYyNzIzMDMtYmEwM2E3M2ItODI5ZC00YjFhLTlmZDAtOWJlZWMzMjg3MDZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDA5NDIwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2NjYxMjY3OTM5NjZmNGFkNjEyZTIxYjcyYjQwMDczOGU4OWIxZGU5YWRhOTA2M2QzNDc3MGYwNmRjODVlMGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.i7U-J9l8XYyElQmWpRhF0S6WogY3CCjavVP70R00P8U)

@MeganSteenkamp I have updated the flows and added screenshots if you could re review

MeganSteenkamp commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -34,24 +34,27 @@ sequenceDiagram
-			POS->>Centrapay: POS Accepts Condition
-		else Payment Request Status is paid
-			Note over POS: Stop Polling for Confirmation
+			POS Server->>Centrapay: Accept Condition
```
Change: This is missing an arrow where the cashier would accept the condition on the POS, then it goes to the POS server.

MeganSteenkamp commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -382,41 +386,48 @@ autonumber
-		end
+  opt Payment Condition Pending
+    note over POS: ❓ Prompt Cashier to Accept or Decline Condition
+    POS->>Centrapay: POS Accepts Condition
```
Change: In your other diagram you have this POS server -> Centrapay. Change to make it consistent to go POS -> POS Server -> Centrapay.

MeganSteenkamp commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -382,41 +386,48 @@ autonumber
-
+  POS->>POS Server: Request Confirmation
+	POS Server->>Centrapay: Make Confirmations Against Authorised Funds
+  POS Server->>POS: Send Confirmation Status
```
Question: Is the POS server pushing a notification here or is it the response to line 415? If it is the response, I think this would be better like:

```
POS->>POS Server: Make Confirmations Against Authorised Funds
POS Server->>Centrapay: Make Confirmations Against Authorised Funds
Centrapay-->>POS Server: Response
POS Server-->>POS: Response
```

MeganSteenkamp commented on src/content/guides/farmlands-pos-integration.mdoc:
```diff
@@ -382,41 +386,48 @@ autonumber
+  POS->>POS Server: Requst Release of Authorised Funds
+  POS Server->>Centrapay: Release Remaining Authorised Funds
+  POS Server->>POS: Send Release Status
+  Note over POS: ✅ Display Release confirmation
```
Same here as comment above.

MeganSteenkamp reviewed (approved):
Approved with comments ✅
